libexec/bats-core/bats-exec-file

```diff
@@ -222,9 +222,16 @@ bats_run_tests_in_parallel() {
       ((++test_number_in_suite))
       ((++test_number_in_file))
       mkdir -p "$output_folder/$test_number_in_suite"
-      bats_semaphore_run "$output_folder/$test_number_in_suite" \
-                          "$BATS_LIBEXEC/bats-exec-test" "${flags[@]}" "$filename" "$test_name" "$test_number_in_suite" "$test_number_in_file" \
-                          > "$output_folder/$test_number_in_suite/pid"
+      # deal with empty flags to avoid spurious "unbound variable" errors on Bash 4.3 and lower
+      if [[ "${#flags[@]}" -gt 0 ]]; then
```

Comment:
> Have a look at bats-exec-suite for an example that does not need to duplicate the code. In essence: flags is initialized with --dummy-flag, which is then ignored by the callee.
@martin-schulze-vireso I was looking at that as an alternative, but obviously couldn't connect all the dots. Anyway, I updated PR accordingly.

> Thanks for the PR. I am suprised that this slipped through, as there are tests in place for exactly this scenario. 

AFAICT, the preconditions for this to trigger are:
  - bash version < 4.4
  - parallel run with --jobs > 1
  - `load()`-ed helper does `set -u`

> Can you provide a reproducer?

I'll try to add a test case for this.

Reply:
@martin-schulze-vireso I added test cases for both helper and test doing 'set -u'. The test `skip()` if they are run with too new bash. On old bash the test fail without `a289834ca3a3` and pass otherwise.